api/dps/server_integration_test.go

```diff
@@ -0,0 +1,1024 @@
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.First(mocks.GenericHeight))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetFirstRequest{}
+		resp, err := server.GetFirst(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeight, resp.Height)
+	})
+
+	t.Run("handles indexer failure on First", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetFirstRequest{}
+		_, err = server.GetFirst(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_GetLast(t *testing.T) {
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Last(mocks.GenericHeight))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetLastRequest{}
+		resp, err := server.GetLast(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeight, resp.Height)
+	})
+
+	t.Run("handles indexer failure on Last", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetLastRequest{}
+		_, err = server.GetLast(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_GetHeightForBlock(t *testing.T) {
+	id := mocks.GenericIdentifier(0)
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Height(id, mocks.GenericHeight))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetHeightForBlockRequest{
+			BlockID: id[:],
```

Comment:
I'm on the verge myself. Buut, as I commented on a previous PR before, since we've had numerous occasions of slicing a loop `flow.Identifier` variable that introduces bugs, I don't think it's a terrible idea to have something like that, that can be used as a rule.

Reply:
IMO in cases where it's straight forward, I'd rather use slicing directly. Always using the convert function, even when it's not necessary, might be a bad habit. It might confuse people outside of the project into thinking that slicing is not allowed for some weird reason. If we only use it when it makes sense, the purpose of the func is clearer.